types/systemjs/index.d.ts

```diff
@@ -1,388 +1,103 @@
-         * Synchronous alternative to `SystemJS.resolve`.
-         */
-        resolveSync(moduleName: string, parentName?: string): string;
-
-        /**
-         * In CommonJS environments, SystemJS will substitute the global require as needed by the module format being
-         * loaded to ensure the correct detection paths in loaded code.
-         * The CommonJS require can be recovered within these modules from System._nodeRequire.
-         */
-        _nodeRequire(dep: string): any;
-
-        /**
-         * Modules list available only with trace=true
-         */
-        loads: PackageList<any>;
-
-        env: string;
-
-        loaderErrorStack: boolean;
+    /**
+     * Delete a module from the module registry. Note that the moduleId almost always must be a full url and that
+     * you might need to call System.resolve() to obtain the moduleId for modules in an import map.
+     * The returned function is intended for use after re-importing the module. Calling the function
+     * will re-bind all the exports of the re-imported module to every module that depends on the module.
+     */
+    delete(moduleId: string): false | typeof UpdateModuleFn;
 
-        packageConfigPaths: string[];
+    /**
+     * Get a module from the SystemJS module registry. Note that the moduleId almost always must be a full url
+     * and that you might need to call System.resolve() to obtain the moduleId. If the module does not exist in
+     * the registry, null is returned.
+     */
+    get(moduleId: string): Module | null;
+    // tslint:disable-next-line no-unnecessary-generics
+    get<T>(moduleId: string): T | null;
 
-        /**
-         * Specify a value of true to have SystemJS conform to the AMD-style plugin syntax, e.g. "text!some/file.txt", over the default of "some/file.txt!text".
-         */
-        pluginFirst: boolean;
+    /**
+     * Indicates whether the SystemJS module registry contains a module. Note that the moduleId almost always
+     * must be a full url and that you might need to call System.resolve() to obtain the moduleId.
+     */
+    has(moduleId: string): boolean;
 
-        version: string;
+    /**
+     * An alternative to System.register(), this allows you to insert a module into the module registry. Note that
+     * the moduleId you provide will go straight into the registry without being resolved first.
+     */
+    set(moduleId: string, module: Module): void;
 
-        /**
-         * Enables the output of warnings to the console, including deprecation messages.
-         */
-        warnings: boolean;
-    }
+    /**
+     * Use for (let entry of System.entries()) to access all of the modules in the SystemJS registry.
+     */
+    entries(): Iterable<typeof ModuleEntry>;
+  }
+
+  // tslint:disable-next-line no-unnecessary-generics
+  function ImportFn<T extends Module>(moduleId: string, parentUrl?: string): Promise<T>;
+
+  function DeclareFn(_export: typeof ExportFn, _context: Context): Declare;
```

Comment:
What are all these uppercase-named functions for? Near as I can tell, there's no `System.DeclareFn` in the public API?

Reply:
They are for reuse within this file and are not meant to be exported. I noticed that they were being exported but didn't know how to implement types, interfaces, and functions in a global typescript declaration without all of them being exported. I tried to follow [this pattern](https://www.typescriptlang.org/docs/handbook/namespaces.html#working-with-other-javascript-libraries) from the typescript handbook.

Note that my types here have a `SystemJS` namespace and then a `System` const. Newly with systemjs@6, there is actually no longer a `SystemJS` global variable at all. The only global variable is `System`.

Reply:
Add `export {}` into the namespace declaration - then only other members with an `export` modifier will be exported. But if you's just using these to `typeof` them later, it's probably better if you just used type aliases, too, eg `type ImportFn = <T extends Module>(moduleId: string, parentUrl?: string) => Promise<T>;`

Reply:
👍 thanks I will update this.